### PR TITLE
Nurse reports charting

### PR DESCRIPTION
**Pull request by divs1210**

### CODE REVIEW

File: app/queries/patients_query.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class PatientsQuery
+  attr_reader :relation
+
+  def initialize(relation = Patient.all)
+    @relation = relation
+  end
+
+  def registered_at(facility_id)
+    relation
+      .where("registration_facility_id = '#{facility_id}'")
```
Comment (govindkrjoshi) [inline app/queries/patients_query.rb]:
```suggestion
      .where(registration_facility_id: facility_id)
```

File: app/controllers/api/current/analytics/user_analytics_controller.rb [hunk 2]
```diff
@@ -18,23 +14,24 @@ def show
...
-  def new_patients_by_facility_week
-    Facility.joins(:patients) \
-      .where("facilities.id = '#{current_facility.id}'") \
-      .distinct('patients.id') \
-      .group_by_week('patients.device_created_at', last: @weeks_previous) \
-      .count
-  end
-
+  # dev only
   def mock_new_patients_by_facility_week
```
Comment (govindkrjoshi) [inline app/controllers/api/current/analytics/user_analytics_controller.rb]: We can remove this method now.

File: spec/queries/patients_query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+RSpec.describe PatientsQuery, type: :query do
+  let!(:this_facility) { FactoryBot.create(:facility) }
+  let!(:other_facility) { FactoryBot.create(:facility) }
+  let!(:patient_in_this_facility) { FactoryBot.create(:patient, registration_facility: this_facility) }
+  let!(:patient_in_other_facility) { FactoryBot.create(:patient, registration_facility: other_facility) }
+
+  it 'should return only patients registered in this_facility' do
+    patients_in_this_facility = PatientsQuery.new.registered_at(this_facility.id)
+    expect(patients_in_this_facility.count).to eq(1)
+    expect(patients_in_this_facility.first).to eq(patient_in_this_facility)
```
Comment (govindkrjoshi) [inline spec/queries/patients_query_spec.rb]: I think it's better to assert for inclusion here.

File: app/controllers/api/current/analytics/user_analytics_controller.rb (hunk 2 above)
Comment (divs1210) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: removed

File: spec/queries/patients_query_spec.rb (hunk 3 above)
Comment (divs1210) [reply spec/queries/patients_query_spec.rb]: done

File: app/queries/patients_query.rb (hunk 1 above)
Comment (divs1210) [reply app/queries/patients_query.rb]: done

**Pull request merged**